File: projects/10/java/Flying.java
/* Flying.java provides a Flying subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Ian Christensen, for CS 214 at Calvin College
 * Date: Spring, 2018
 */

public class Flying extends Bird {
    
	public Flying() {
		super();
	}
	
	/* explicit constructor
	 * Receive: name, a String 
	 * PostCond: myName == name.
	 */
	public Flying(String name) {     
		super(name);
	} 

	/* A Flying's move
	 * Return: a Flying-move ("flew passed").
	 */
	public String move() {
		return "flew passed";
	} 
}

File: projects/10/ruby/Flying.rb
#!/usr/bin/ruby
# Flying.rb defines a flying superclass to be extended by specific bird sub-classes
# Begun by: Dr. Adams, for CS 214 at Calvin College
# Completed by: Ian Christensen, for CS 214 at Calvin College
# Date: Spring, 2018

require './Bird.rb'

class Flying < Bird
	
	# Method: move, gets the movement of the bird
	# Precondition: Bird must exist
	# Receives: a Bird object
	# Returns: 'flew passed', the generic movement
	def move
		'flew passed'
	end
end
